### PR TITLE
ExpDecay start step

### LINKED ISSUE #1815: Can ExpDecay of learning rate start at some intermediate step?
I would like to start training a model with a constant learning rate (LR) first, and then after some epochs, start decaying exponentially the LR. I think this is not an uncommon practice.

It would be nice to add this option to `ExpDecay`.

### PR DESCRIPTION
**Pull request by cossio**
Adds an option to `ExpDecay` which specifies the step at which the exponential decay of the learning rate starts.

Fixes #1815.

### PR Checklist

- [x] Tests are added
- [ ] Entry in NEWS.md
- [x] Documentation, if applicable
- [ ] API changes require approval from a committer (different from the author, if applicable)

### CODE REVIEW

File: src/optimise/optimisers.jl [hunk 1]
```diff
@@ -624,16 +625,17 @@ mutable struct ExpDecay <: AbstractOptimiser
...
 end
 
-ExpDecay(opt = 0.001, decay = 0.1, decay_step = 1000, clip = 1e-4) = 
-  ExpDecay(opt, decay, decay_step, clip, IdDict())
+ExpDecay(opt = 0.001, decay = 0.1, decay_step = 1000, clip = 1e-4, start = 1) =
+  ExpDecay(opt, decay, decay_step, clip, start, IdDict())
 
 function apply!(o::ExpDecay, x, Δ)
-  η, s, decay = o.eta, o.step, o.decay
+  η, s, decay, n0 = o.eta, o.step, o.decay, o.start
```
Comment (DhairyaLGandhi) [inline src/optimise/optimisers.jl:636]: `n0 -> start` for consistency?

**Review by DhairyaLGandhi — COMMENTED** [review]
Mostly lgtm.

File: src/optimise/optimisers.jl [hunk 2]
```diff
@@ -624,16 +625,17 @@ mutable struct ExpDecay <: AbstractOptimiser
...
   n = o.current[x] = get(o.current, x, 0) + 1
-  if o.current[x]%s == 0 && count(x -> x%s == 0, values(o.current)) == 1
+  if n ≥ n0 && (n - n0 + 1) % s == 0 && count(x -> (x - n0 + 1) % s == 0, values(o.current)) == 1
```
Comment (DhairyaLGandhi) [inline src/optimise/optimisers.jl:638]: We can get rid of `current` if it's not needed anymore

File: src/optimise/optimisers.jl (hunk 1 above)
Comment (cossio) [reply src/optimise/optimisers.jl]: agreed.

File: src/optimise/optimisers.jl (hunk 2 above)
Comment (cossio) [reply src/optimise/optimisers.jl]: I don't understand. `current` is needed.

Comment (DhairyaLGandhi) [reply src/optimise/optimisers.jl]: So it is, my bad!

**Comment by cossio** [comment]
@DhairyaLGandhi merge?

**Review by DhairyaLGandhi — APPROVED** [review]
bors r+

**Comment by cossio** [comment]
@DhairyaLGandhi  Can you retrigger bors?

**Review by DhairyaLGandhi — APPROVED** [review]
bors r+

**Review by darsnack — APPROVED** [review]
